core/src/avm2/verify.rs

```diff
@@ -112,28 +112,41 @@ pub fn verify_method<'gc>(
             let bytes_read = new_position - previous_position;
             assert!(bytes_read > 0);
 
-            byte_info.insert(previous_position, ByteInfo::OpStart(op.clone()));
+            byte_info[previous_position as usize] = ByteInfo::OpStart(op.clone());
             for j in 1..bytes_read {
-                byte_info.insert(previous_position + j, ByteInfo::OpContinue);
+                byte_info[(previous_position + j) as usize] = ByteInfo::OpContinue;
             }
 
-            let mut check_target = |offs: i32, is_jump: bool| {
+            let mut check_target = |seen_targets: &HashSet<i32>, offs: i32, is_jump: bool| {
                 let target_position = if is_jump {
                     offs + new_position
                 } else {
                     offs + previous_position
                 };
 
-                let hash_map_info = byte_info.get(&target_position);
+                let lookedup_target_info = byte_info.get(target_position as usize);
 
-                if matches!(hash_map_info, Some(ByteInfo::OpContinue)) {
+                if matches!(lookedup_target_info, Some(ByteInfo::OpContinue)) {
                     return Err(make_error_1021(activation));
                 }
 
                 if target_position < 0 || target_position as usize >= body.code.len() {
                     return Err(make_error_1021(activation));
                 }
 
+                // Ensure backwards jumps to not-yet-jumped-to code target a `Label` op
+                if !seen_targets.contains(&target_position) && offs < 0 {
+                    reader.seek_absolute(&body.code, target_position as usize);
```

Comment:
random thought: if the jump is backwards, can we guarantee that all the ops before it were already seen? If so, then `byte_info` should be reached and we could use that instead?
That said, the current impl should work too and I don't wanna rock the boat too much

Reply:
> if the jump is backwards, can we guarantee that all the ops before it were already seen?

No, definitely not, since there can be gaps in bytecode.